### ISSUE TITLE
RetrieveRequestExecutor does not return correct response

### ISSUE THREAD

**Issue by Zerajima** [DynamicsValue/fake-xrm-easy#142]
In response returned by [RetrieveRequestExecutor.Execute](https://github.com/DynamicsValue/fake-xrm-easy-core/blob/179a9fe4b8a5c99098da464d17f077b87a2627db/src/FakeXrmEasy.Core/Middleware/Crud/FakeMessageExecutors/RetrieveRequestExecutor.cs#L79C56-L79C62) method, the entity object is stored under "Entity" key. 

This is not the same key, as returned by dataverse. Instead "BusinessEntity" is used. This behavior is described in https://learn.microsoft.com/en-gb/power-apps/developer/data-platform/understand-the-data-context#outputparameters.

I ran into this issue when writing tests for plugin I wrote, for Retrieve message.

**Comment by BetimBeja** [comment]
Hi @Zerajima 
I think you are getting confused between a plugin registered in the `Retrieve` message which contains the `BusinessEntity` output variable, and the `RetrieveResponse` type used from the SDK in combination with `RetrieveRequest`. The `RetrieveResponse` has an `Entity` parameter. See here https://learn.microsoft.com/en-us/dotnet/api/microsoft.xrm.sdk.messages.retrieveresponse

I am not sure if @jordimontana82 has implemented something that triggers plugins on the `Retrieve` message, or `RetrieveMultiple` message, he can reply here if that is the case :)

**Comment by Zerajima** [comment]
You're correct, it's pretty confusing, and we have Microsoft to that for that. RetrieveResponse does actually contain "Entity" field, however during the request execution pipeline, this value is stored in IExecutionContext.OutputParameters, under "BusinessEntity" key. The link I provided in first post describes the differences between message responses and IExecutionContext contents. There's a list of responses, that use different naming in response class and in IExecutionContext.

I'm not sure where exactly this is handled in FakeXrmEasy, I assumed it would be in message executor, but if it's somewhere else, I apologize for misleading you.

This is the code snippet, that works correctly on dataverse, but throws KeyNotFound exception when executed through FakeXrmEasy pipeline.
`var resultEntity = context.OutputParameters["BusinessEntity"] as Entity;`

**Comment by BetimBeja** [comment]
In `FakeXrmEasy` the `MessageExecutor` like `RetrieveRequestExecutor` and the `EventPipeline` Simulation are two different things. I guess this might be picked up from @jordimontana82 as an improvement.

**Comment by jordimontana82** [comment]
> I am not sure if @jordimontana82 has implemented something that triggers plugins on the Retrieve message, or RetrieveMultiple message, he can reply here if that is the case :)

The pipeline should support any message (including Retrieve's). 

@Zerajima can you post a sample unit test to reproduce, please?

FXE (short for FakeXrmEasy) passes in InputParameters from the incoming request and out to OutputParameters as well. But a a  a unit test might be handy (along with your pipeline setup and how you are registering plugin steps) to reproduce in case we missed anything...

**Comment by jordimontana82** [comment]
Maybe it's related to this one too #124

**Comment by Zerajima** [comment]
Here's a sample test to demonstrate this issue. The IPlugin implementation should work properly on dataverse.

```
using FakeXrmEasy.Abstractions.Enums;
using FakeXrmEasy.Abstractions.Plugins.Enums;
using FakeXrmEasy.Middleware;
using FakeXrmEasy.Middleware.Crud;
using FakeXrmEasy.Middleware.Pipeline;
using FakeXrmEasy.Pipeline;
using FakeXrmEasy.Plugins.PluginSteps;
using Microsoft.Xrm.Sdk.Client;
using Microsoft.Xrm.Sdk.Extensions;
using Microsoft.Xrm.Sdk.Query;
using Xunit;

namespace Tests;

public class SampleTests
{
    private const string EntityLogicalName = "testentity";
    private const string NameAttributeLogicalName = "name";

    [Fact]
    public void TestPlugin()
    {
        var context = MiddlewareBuilder
            .New()
            .AddCrud()
            .AddPipelineSimulation()
            .UsePipelineSimulation()
            .UseCrud()
            .SetLicense(FakeXrmEasyLicense.NonCommercial)
            .Build();

        context.RegisterPluginStep<PluginHandler>(
            new PluginStepDefinition()
            {
                MessageName = "Retrieve",
                EntityLogicalName = EntityLogicalName,
                Stage = ProcessingStepStage.Postoperation,
                Mode = ProcessingStepMode.Synchronous,
                Rank = 1,
            });

        var seededEntity = new TestEntity()
        {
            Id = Guid.NewGuid(),
            Name = "Seeded name",
        };
        context.AddEntity(seededEntity);

        var service = context.GetOrganizationService();
        var actualEntity = service.Retrieve(EntityLogicalName, seededEntity.Id, new ColumnSet(NameAttributeLogicalName));

        Assert.Equal(actualEntity.Attributes[NameAttributeLogicalName], "Modified name");
    }

    [EntityLogicalName(EntityLogicalName)]
    public class TestEntity : Entity
    {
        public TestEntity()
            : base(EntityLogicalName)
        {
        }

        [AttributeLogicalName(NameAttributeLogicalName)]
        public string? Name
        {
            get => GetAttributeValue<string?>(NameAttributeLogicalName);
            set => SetAttributeValue(NameAttributeLogicalName, value);
        }
    }

    public class PluginHandler : IPlugin
    {
        public void Execute(IServiceProvider serviceProvider)
        {
            var context = serviceProvider.Get<IPluginExecutionContext>();

            var entity = context.OutputParameters["BusinessEntity"] as Entity; // Throws KeyNotFoundException
            entity.Attributes[NameAttributeLogicalName] = "Modified name";
        }
    }
}
```
